framework/src/fvkernels/FVFluxKernel.C

```diff
@@ -327,8 +327,21 @@ FVFluxKernel::gradUDotNormal() const
   const bool correct_skewness =
       (_var.faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage);
 
+  const Elem * const elem = &_face_info->elem();
+  const Elem * const neighbor = _face_info->neighborPtr();
+
+  bool var_defined_on_elem = _face_info->varDefinedOnElem(_var.name());
+  bool is_internal_face = _var.isInternalFace(*_face_info);
+
+  const ADReal side_one_value = (!is_internal_face && !var_defined_on_elem)
+                                    ? _var.getBoundaryFaceValue(*_face_info)
+                                    : _var.getElemValue(elem);
```

Comment:
if internal face and variable not defined on element, can we take an element value?